.github/workflows/release-shell-pkg.yaml

```diff
@@ -0,0 +1,46 @@
+name: Publish Shell Package
+
+on:
+  push:
+    tags:
+        - 'shell-pkg-v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version: '16.x'
+
+    - name: Validate Plugin build system
+      run: ./shell/scripts/test-plugins-build.sh
+      shell: bash
+
+    - name: Reset repository (file system)
+      run: | 
+        git reset --hard HEAD
+        echo $YARN_REGISTRY
+        cat ./shell/package.json
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version: '16.x'
+        registry-url: 'https://registry.npmjs.org'
+        scope: '@rancher'
+    
+    - name: Install packages
+      run: yarn install
```

Comment:
I think that we should be using `yarn install --frozen-lockfile` in ci. 

> If you need reproducible dependencies, which is usually the case with the continuous integration systems, you should pass --frozen-lockfile flag.

Reply:
updated. Good one